src/main/java/agh/ics/oop/RectangularMap.java

```diff
@@ -0,0 +1,64 @@
+package agh.ics.oop;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class RectangularMap implements IWorldMap{
+    private int width;
+    private int height;
+
+    private List<Animal> animals;
+
+
+    RectangularMap(int width, int height){
+        this.width = width;
+        this.height = height;
+        this.animals = new ArrayList<>();
+    }
+
+    @Override
+    public boolean canMoveTo(Vector2d position) {
+        return (position.x >= 0 && position.x <= this.width && position.y >= 0 && position.y <= this.height) && ! isOccupied(position);
+    }
+
+    @Override
+    public boolean place(Animal animal) {
+        ///?????????????
+        //umożliwia występowanie więcej niż jednego zwierzęcia na mapie,
```

Comment:
a to fakt ;) ten problem będziemy jeszcze rozwiązywać, ale gdy wprowadzimy nieco wydajniejszą implementację mapy